### PR TITLE
feat(templates): update Cloudflare templates to use `wrangler` v2

### PR DESCRIPTION
**Pull request by aiji42**
Update template for cloudflare workers. (https://github.com/remix-run/remix/discussions/3158)
- `wrangler.toml`: Changed to wrangler v2 style.
- `package.json`: Added wrangler to dependecncies. Updated @cloudflare/workers-types latest. (also pages template)
- `remix.env.d.ts`: Fixed dead link. (also pages template)
- `README`: Removed the sentence about installation of wrangler.

I have confirmed that it is deployable. (https://github.com/aiji42/remix-cf-wrangler-2)
```
uejima.aiji$ yarn deploy                                                                                             [~/git/remix-cf-wrangler-2]
yarn run v1.22.18
$ npm run build && wrangler publish

> build
> remix build

Building Remix app in production mode...
Built in 385ms
 ⛅️ wrangler 2.0.2
-------------------
Running custom build: npm run build

> build
> remix build

Building Remix app in production mode...
Built in 368ms
Reading public/build/_shared/chunk-LHODYXPX.js...
Skipping - already uploaded.
Reading public/build/_shared/chunk-OGYP3M3B.js...
Skipping - already uploaded.
Reading public/build/entry.client-GY23DEVV.js...
Skipping - already uploaded.
Reading public/build/manifest-289EFF28.js...
Skipping - already uploaded.
Reading public/build/root-NVPFOTJV.js...
Skipping - already uploaded.
Reading public/build/routes/index-UD6EN4U7.js...
Skipping - already uploaded.
Reading public/favicon.ico...
Skipping - already uploaded.
↗️  Done syncing assets
Uploaded remix-cf-wrangler-2 (2.18 sec)
Published remix-cf-wrangler-2 (0.31 sec)
  remix-cf-wrangler-2.aiji422990.workers.dev
✨  Done in 7.46s.
```

### CODE REVIEW

File: templates/cloudflare-pages/package.json [hunk 1]
```diff
@@ -20,7 +20,7 @@
     "react-dom": "^17.0.2"
   },
   "devDependencies": {
-    "@cloudflare/workers-types": "^3.4.0",
+    "@cloudflare/workers-types": "^3.10.0",
```
Comment (MichaelDeBoey) [inline templates/cloudflare-pages/package.json:23]: I think we should update `wrangler` dependency here as well?

File: templates/cloudflare-workers/wrangler.toml [hunk 2]
```diff
@@ -1,17 +1,13 @@
 name = "remix-cloudflare-workers"
-type = "javascript"
 
-zone_id = ""
+compatibility_date = "2022-05-11"
+
```
Comment (MichaelDeBoey) [inline templates/cloudflare-workers/wrangler.toml:4]: I think this one can be omitted, no?

```suggestion
```

**Review by MichaelDeBoey — CHANGES_REQUESTED** [review]

**Comment by MichaelDeBoey** [comment]
@jacob-ebey Is there a reason why `cloudflare-pages` template doesn't have a `wrangler.toml` file? 🤔
It seems odd that we have one in `cloudflare-workers` but not in cloudflare-pages`, even though we have `wrangler` in `devDependencies`

File: templates/cloudflare-workers/wrangler.toml (hunk 2 above)
Comment (aiji42) [reply templates/cloudflare-workers/wrangler.toml:4]: Specifying `compatibility_date` is required.
```
> build
> remix build

Building Remix app in production mode...
Built in 370ms
 ⛅️ wrangler 2.0.2
-------------------
Running custom build: npm run build

> build
> remix build

Building Remix app in production mode...
Built in 286ms

✘ [ERROR] A compatibility_date is required when publishing. Add one to your wrangler.toml file, or pass it in your terminal as --compatibility-date. See https://developers.cloudflare.com/workers/platform/compatibility-dates for more information.


If you think this is a bug then please create an issue at https://github.com/cloudflare/wrangler2/issues/new.
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

File: templates/cloudflare-pages/package.json (hunk 1 above)
Comment (aiji42) [reply templates/cloudflare-pages/package.json:23]: I will fix it.

Comment (aiji42) [reply templates/cloudflare-pages/package.json:23]: Fixed!

**Review by MichaelDeBoey — APPROVED** [review]

**Comment by himorishige** [comment]
Thank you for providing a useful template.

One suggestion would be.

The kv-asset-handler installed from the template seemed to be v0.1.3.
Since v0.2.0, some definitions have changed, such as `getAssetFromKV`, etc.
I would like to see kv-asset-handler also bundled with v0.2.0 or higher.

https://github.com/cloudflare/kv-asset-handler/releases/tag/v0.2.0

**Comment by MichaelDeBoey** [comment]
@aiji42 Could you please rebase this PR onto latest `main` & resolve conflicts?

**Comment by acusti** [comment]
@aiji42 thanks for putting this together! i just applied your changes manually (updating `wrangler.toml` and adding `wrangler` v2.0.5 as a dev dependency) and it works perfectly.

-----------

for anyone else running into issues and searching for answers, i originally tried to create a new remix project using `cloudflare-workers` as the build target, and when i installed `wrangler` using the instructions in the doc that the remix template’s README links to and ran `npm run deploy`, i got the following output:

```
 ⛅️ wrangler 2.0.5
-------------------
▲ [WARNING] Processing wrangler.toml configuration:

    - 😶 Ignored: "type":
      Most common features now work out of the box with wrangler, including modules, jsx,
  typescript, etc. If you need anything more, use a custom build.
    - Deprecation: "zone_id":
      This is unnecessary since we can deduce this from routes directly.
    - Deprecation: "build.upload.format":
      The format is inferred automatically from the code.
    - Deprecation: "site.entry-point":
      Delete the `site.entry-point` field, then add the top level `main` field to your configuration
  file:
      ```
      main = "index.js"
      ```


✘ [ERROR] Processing wrangler.toml configuration:

    - Expected "route" to be either a string, or an object with shape { pattern, zone_id | zone_name
  }, but got "".
```

replacing the contents of `wrangler.toml` with this will make it work:
```toml
name = "remix-cloudflare-workers"

compatibility_date = "2022-05-11"

account_id = ""
workers_dev = true
main = "./build/index.js"

[site]
bucket = "./public"

[build]
command = "npm run build"
```

**Comment by himorishige** [comment]
@aiji42 

Thanks for your feedback. As you indicated, it may not necessarily need to be updated as a template. We will withdraw this proposal.
`@cloudflare/kv-asset-handler` is used to create a Worker environment that is an adapter instead of `server.js` when using DO. But I don't think that is necessary for the template.

**Comment by zwily** [comment]
Wrangler 2 also includes miniflare, so should this change also remove the explicit miniflare dependency and change the `start` and `dev:miniflare` scripts to call `wrangler dev --local`?

**Comment by zwily** [comment]
> > * Workers Sites projects will still deploy with wrangler2, however, you can no longer create them. We recommend using Cloudflare Pages instead.
> 
> Given that Remix on Workers relies on being a worker site, shouldn't we stick with wrangler 1?

I just tested this - I created a new Workers app with `npx create-remix`, and then applied the changes in this patchset to it to get it on wrangler 2, and it created and deployed the site just fine. So I'm not sure what that note in the Cloudflare Discord could be referring to.

**Comment by terribleplan** [comment]
I arrived at these changes independently of seeing this PR since the CF docs now say to use wrangler2 from NPM. Most things seem to be working except for static asset _serving_.

I can see the assets get uploaded, stored in a KV namespace, and be requested by a client, but something either in `@cloudflare/kv-asset-handler` or how it is called is not working correctly, and I am getting a 404 for files under `/build`.

I tried messing with `publicPath` in `remix.config.js` (as the assets in KV are prefixed with `public/build` and I am unsure how mapping from a request path to KV key works) with no change.

I wonder if this might be related to the "you can no longer create them" mentioned above, as this is a new worker/site, and as far as I can tell everything _should_ be working. Using pages is also sadly a non-starter at the moment as pages does not support R2 bindings yet and the docs poorly explain the new manual upload deployment method (as this project isn't in github/gitlab).


It is also worth noting that I had to manually resolve miniflare (when using yarn at least) as different versions were being installed to satisfy the direct dev dependency vs the dependency through wrangler, which was causing weird issues in dev (failing to parse incoming requests).
```
  "resolutions": {
    "miniflare": "^2.5.0"
  }
```

**Comment by pyrossh** [comment]
I got it working with the above instructions on wrangler 2. The images are working fine for me as I import them in code though not served statically.

**Comment by n3oney** [comment]
Just a question. Does the `deploy` script in the package.json actually need the `npm run build` before `wrangler publish`, if we already have a `npm run build` in the build section of the `wrangler.toml`? It seems like it's going to build it twice.

**Comment by john-griffin** [comment]
What is left to do here? Need a hand with anything? Wrangler 2 is the official way forward and I'm seeing a lot of people in the community confused when generating a new app.

**Comment by MichaelDeBoey** [comment]
@aiji42 Could you please rebase your branch on latest `dev` & resolve conflicts?

**Comment by aiji42** [comment]
@MichaelDeBoey
Is it latest `dev`, not latest `main`? I have main as the target branch since it is a template change.
Also, the version of wrangler has been upgraded, so I will update to use the latest.

File: templates/cloudflare-workers/wrangler.toml [hunk 3]
```diff
@@ -1,6 +1,5 @@
 name = "remix-cloudflare-workers"
 
-account_id = ""
```
Comment (aiji42) [inline templates/cloudflare-workers/wrangler.toml:3]: Removed `account_id` due to the following warning during deployment.

```
$ wrangler publish
 ⛅️ wrangler 2.0.22
--------------------
▲ [WARNING] Processing wrangler.toml configuration:

    - The "account_id" field in your configuration is an empty string and will be ignored.
      Please remove the "account_id" field from your configuration.
```

**Comment by aiji42** [comment]
I rebased to the latest main and then resolved the conflicts. I updated to the latest wrangler.

**Review by MichaelDeBoey — APPROVED** [review]
Still wondering if we also need a `wrangler.toml` for `cloudflare-pages` 🤔

CC/ @jacob-ebey

**Comment by aiji42** [comment]
@MichaelDeBoey 
No need for a `wrangler.toml` in `cloudflare-pages` application.
This is because the repositories are linked from the Cloudflare Pages dashboard to build and deploy.
https://developers.cloudflare.com/pages/framework-guides/remix/

**Review by kentcdodds — APPROVED** [review]
Looks good. Thank you!

**Pull request merged**